accounts/abi/bind/base.go

```diff
@@ -411,6 +435,15 @@ func (c *BoundContract) transact(opts *TransactOpts, contract *common.Address, i
 	return signedTx, nil
 }
 
+// Event returns the event with the given name, or error if no such event exists.
+func (c *BoundContract) Event(name string) (*abi.Event, error) {
```

Comment:
Same question as with the ABI, why expose this instead of use higher abstractions?

Reply:
This is more of a convenience. This way a developer can call `b.Event("methodName")` directly from a bound contract, and use the Event object to grab the RawName, parse a specific argument, etc...

Reply:
The same applies to the Method in the template. https://github.com/ethereum/go-ethereum/blob/0f65144fc05d95c8cb25606fd695ee11434f91da/accounts/abi/bind/template.go#L322C3-L322C3